src/index.ts

```diff
@@ -132,15 +134,59 @@ function addSegmentFromDirectionsResponse(previousLngLat: LngLat, e: MapMouseEve
     });
 }
 
-function addSegmentFromStraightLine(previousLngLat: LngLat, e: MapMouseEvent): void {
-  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, e.lngLat);
+function addSegmentFromStraightLine(previousLngLat: LngLat, lngLat: LngLat, animate = true): void {
+  const newSegment = nextSegmentService.segmentFromStraightLine(previousLngLat, lngLat);
 
-  animationService.animateSegment(newSegment);
-  const marker = addMarker(e.lngLat, false);
+  if (animate) animationService.animateSegment(newSegment);
+  const marker = addMarker(lngLat, false);
   currentRun.addSegment(newSegment, marker);
   updateLengthElement();
 }
 
+function runToJson(run: CurrentRun): string {
+  if (run === undefined) return "{}";
+  let runJSON: {[name:string]: any} = {
+    start: {
+      lng: run.start.lngLat.lng,
+      lat: run.start.lngLat.lat,
+    },
+    distance: run.distance,
+    segments: [],
+    followRoads
+  }
+  for (let i in run.segments) {
```

Comment:
also just a suggestion on renaming `i` to `segment` or something, I was initially thinking this was an incrementing loop and not a foreach loop =P

Reply:
it is an incrementing loop. if `in` is changed to of then i should be segment. If you'd prefer that, I can do that